base/strings/util.jl

```diff
@@ -195,6 +195,60 @@ end
 # TODO: optimization for the default case based on
 # chop(s::AbstractString) = SubString(s, firstindex(s), prevind(s, lastindex(s)))
 
+"""
+    chopprefix(s::AbstractString, prefix::AbstractString) -> SubString
+
+Remove the prefix `prefix` from `s`. If `s` does not start with `prefix`, a string equal to `s` is returned.
+
+See also [`chopsuffix`](@ref).
+
+!!! compat "Julia 1.7"
+    This function is available as of Julia 1.7.
+
+# Examples
+```jldoctest
+julia> chopprefix("Hamburger", "Ham")
+"burger"
+
+julia> chopprefix("Hamburger", "hotdog")
+"Hamburger"
+```
+"""
+function chopprefix(s::AbstractString, prefix::AbstractString)
+    if startswith(s, prefix)
+        SubString(s, nextind(s, lastindex(prefix), 1), lastindex(s))
```

Comment:
This may not be correct if `s` and `prefix` are different subtypes of `AbstractString`, since in that case `lastindex(prefix)` may not be a correct index into `s`.

Reply:
Probably we need two methods: one for `AbstractString` arguments, and an optimized method for `Union{String,SubString{String}}` arguments.